### ISSUE TITLE
[Bug] `gate_output_correction` skips the tail chunk for `T > 65536`

### ISSUE THREAD

**Issue by sablin39** [fla-org/flash-linear-attention#906]
### Checklist

- [x] I have checked [FAQs](https://github.com/fla-org/flash-linear-attention/blob/main/FAQs.md) and existing issues for similar problems
- [x] Please report this bug in English to ensure wider understanding and support

### Describe the Bug

`fla.ops.rwkv7.gate_output_correction` produces incorrect output once the sequence length crosses the 65536-token chunk boundary set by its internal mask. 

Feed it with a sequence with `T>65536`: 
- The `T<=65536` part matches `gate_output_correction_ref`'s output 
- `T>65536` part diverges.

### Steps to Reproduce the Bug

```python
import torch
from fla.ops.rwkv7.gate_output_correction import gate_output_correction, gate_output_correction_ref


B, H, D = 1, 2, 8
dtype = torch.bfloat16
device = "cuda"


def inputs(T):
    o = torch.ones(B, T, H * D, device=device, dtype=dtype)
    r = torch.ones(B, T, H, D, device=device, dtype=dtype)
    k = torch.ones_like(r)
    r_k = torch.ones(H, D, device=device, dtype=dtype)
    v = torch.ones_like(r)
    g = torch.ones_like(o)
    return o, r, k, r_k, v, g


args = inputs(65_536)
assert torch.allclose(gate_output_correction(*args), gate_output_correction_ref(*args)) #works well

args = inputs(65_537)
out = gate_output_correction(*args)
ref = gate_output_correction_ref(*args)
torch.cuda.synchronize()

assert torch.allclose(out[:, :65_536], ref[:, :65_536])
assert torch.allclose(out[:, -1:], ref[:, -1:]) # AssertionError triggers here
```

### Expected Behavior

The code segment is supposed to run with no `AssertionError`. 

### Environment Information

1. Torch: 2.11.0+cu130
2. Triton: 3.6.0
